app/Console/Commands/VariantFilesize.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Models\SizeVariant;
+use Illuminate\Console\Command;
+
+class VariantFilesize extends Command
+{
+	/**
+	 * The name and signature of the console command.
+	 *
+	 * @var string
+	 */
+	protected $signature = 'lychee:variant_filesize {limit=50 : number of photos to process at once (0 means all)}';
+
+	/**
+	 * The console command description.
+	 *
+	 * @var string
+	 */
+	protected $description = 'Set filesize of size variants if missing';
+
+	/**
+	 * Execute the console command.
+	 *
+	 * @return mixed
+	 */
+	public function handle()
+	{
+		$limit = intval($this->argument('limit'));
+
+		if ($this->confirm('This command can take a long time for large instances. Do you really want to run it now ?')) {
+			$variants_query = SizeVariant::query()
+				->where('filesize', '=', -1)->orderBy('id');
+
+			$count = $variants_query->count();
+			if ($count == 0) {
+				$this->line('All filesize variants already set in database.');
+
+				return false;
+			}
+
+			// Number of queries
+			$pages = ceil($count / $limit);
+
+			$bar = $this->output->createProgressBar($count);
+			$bar->start();
+
+			for ($i = 0; $i < $pages; $i++) {
+				// No need to offset, because previous variants have no -1 filesize anymore
+				$variants = $variants_query->limit($limit)->get();
+
+				/* @var SizeVariant $variant */
+				foreach ($variants as $variant) {
+					$fullPath = $variant->full_path;
```

Comment:
Please don't use the attribute `full_path` of size variant in new code. I know it is still there for legacy reasons. Using `full_path` will make the upcoming work of this other guy which wants to implement support for other than local storage only more difficult.

At least use the following code `$fullPath = $variant->getFile()->getAbsolutePath()`. `getFile()` returns a `FlysystemFile` object. Although this code is slightly better, it is still not optimal, because `file_exist($fullPath)` and `filesize($fullPath)` won't work if `getAbsolutePath` might return a non-local path in the future (but maybe a AWS S3 bucket ID).

So, optimally you would use `$variant->getFile()->getSize()` and not use any path at all. However, I know that the method `getSize` does not exist yet and still has to be implemented in a way which uses the Flysystem API. So I won't ask you to do that. But at least do step 1.

Reply:
See 1128a51b.

Reply:
plus, didn't know Flysystem, seems cool :)